trustfall_stubgen/src/root.rs

```diff
@@ -400,3 +403,100 @@ fn make_vertex_file(
+
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, serde::Deserialize)]
+    struct ResultRow {
+        name: String,
+    }
+
+    let mut rows: Vec<_> = trustfall::execute_query(querying_schema, adapter, query, variables)
+        .expect("invalid query")
+        .map(|x| x.try_into_struct::<ResultRow>().expect("invalid conversion"))
+        .collect();
+    rows.sort_unstable();
+
+    let mut uniq: HashMap<String, String> = HashMap::new();
+
+    for row in rows {
+        let name = row.name.clone();
+        let converted = escaped_rust_name(to_lower_snake_case(&name));
+        let v = uniq.insert(converted, name);
+        if let Some(v) = v {
+            panic!(
+                "cannot generate adapter for a schema containing both '{}' and '{}' vertices, consider renaming one of them",
+                v, &row.name
+            );
+        }
+    }
+}
+
+fn ensure_no_edge_name_keyword_conflicts(
+    querying_schema: &Schema,
+    adapter: Arc<SchemaAdapter<'_>>,
+) {
+    let query = r#"
+{
+    VertexType {
+        name @output
+        edge_: edge @fold {
+            names: name @output
+        }
+        property_: property @fold {
+            names: name @output
+        }
+    }
+}"#;
+    let variables: BTreeMap<String, String> = Default::default();
+
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, serde::Deserialize)]
+    struct ResultRow {
+        name: String,
+        edge_names: Vec<String>,
+        property_names: Vec<String>,
+    }
+
+    let mut rows: Vec<_> = trustfall::execute_query(querying_schema, adapter, query, variables)
+        .expect("invalid query")
+        .map(|x| x.try_into_struct::<ResultRow>().expect("invalid conversion"))
+        .collect();
+    rows.sort_unstable();
+
+    for row in &rows {
+        let mut uniq: HashMap<String, String> = HashMap::new();
+
+        for edge_name in &row.edge_names {
+            let edge_name_for_map = edge_name.clone();
+            let converted = escaped_rust_name(to_lower_snake_case(&edge_name_for_map));
+            let v = uniq.insert(converted, edge_name_for_map);
+            if let Some(v) = v {
+                panic!(
+                    "cannot generate adapter for a schema containing both '{}' and '{}' as field names on vertex '{}', consider renaming one of them",
+                    v, &edge_name, &row.name
+                );
+            }
+        }
+        for property_name in &row.property_names {
+            let property_name_for_map = property_name.clone();
+            let converted = escaped_rust_name(to_lower_snake_case(&property_name_for_map));
+            let v = uniq.insert(converted, property_name_for_map);
+            if let Some(v) = v {
+                panic!(
+                    "cannot generate adapter for a schema containing both '{}' and '{}' as field names on vertex '{}', consider renaming one of them",
+                    v, &property_name, &row.name
+                );
+            }
+        }
```

Comment:
rather than having the same loop twice in a row over two different iterables, zip the iterators together then write the loop once
```suggestion
        for field_name in row.edge_names.iter().chain(row.property_names.iter()) {
            let converted = escaped_rust_name(to_lower_snake_case(field_name));
            if let Some(v) = uniq.insert(converted, field_name.clone()); {
                panic!(
                    "cannot generate adapter for a schema containing both '{}' and '{}' as field names on vertex '{}', consider renaming one of them",
                    v, &field_name, &row.name
                );
            }
        }
```

Reply:
Done!